chia/wallet/cc_wallet/cc_wallet.py

```diff
@@ -356,19 +360,20 @@ async def puzzle_solution_received(self, response: PuzzleSolutionResponse, actio
             await self.wallet_state_manager.action_store.action_done(action_id)
 
     async def get_new_inner_hash(self) -> bytes32:
-        return await self.standard_wallet.get_new_puzzlehash()
+        puzzle = adapt_inner_to_singleton(await self.standard_wallet.get_new_puzzle())
+        return puzzle.get_tree_hash()
 
     async def get_new_inner_puzzle(self) -> Program:
-        return await self.standard_wallet.get_new_puzzle()
+        return adapt_inner_to_singleton(await self.standard_wallet.get_new_puzzle())
```

Comment:
Why are you using the singleton adaptor in the CAT adaption code?

Reply:
Because the cat also has truths.  The function is certainly inappropriately named now.